src/com/company/Product.java

```diff
@@ -0,0 +1,47 @@
+package com.company;
+
+public abstract class Product {
+    private int id;
```

Comment:
w srodowisku produkcyjnym raczej nie uzywamy inta do id, czesciej jest to long, ale najczesciej w porzadnych projektach uzywamy w dzisiejszych czasach UUID https://www.baeldung.com/java-uuid, generujac za pomoca metody statycznej `randomUUID`